src/Gravity/TreeGravity.hh

```diff
@@ -240,6 +240,15 @@ private:
   void deserialize(Cell& cell, std::vector<char>::const_iterator& bufItr, const std::vector<char>::const_iterator& endItr) const;
 };
 
+//------------------------------------------------------------------------------
+// Define our static members.
+//------------------------------------------------------------------------------
+template<typename Dimension> unsigned TreeGravity<Dimension>::num1dbits = 21U;
+template<typename Dimension> uint64_t TreeGravity<Dimension>::max1dKey = 1U << TreeGravity<Dimension>::num1dbits;
+template<typename Dimension> uint64_t TreeGravity<Dimension>::xkeymask = (1U << TreeGravity<Dimension>::num1dbits) - 1U;
+template<typename Dimension> uint64_t TreeGravity<Dimension>::ykeymask = TreeGravity<Dimension>::xkeymask << TreeGravity<Dimension>::num1dbits;
+template<typename Dimension> uint64_t TreeGravity<Dimension>::zkeymask = TreeGravity<Dimension>::ykeymask << TreeGravity<Dimension>::num1dbits;
+
```

Comment:
Yeah, that's what I thought, but this was required by Clang 9.0.  I looked up the rules, and it turns out we need them in the header because if you have static members whose values depend on the template parameter they have to be declared in the header so every compilation unit knows the correct value for the template parameter being used.  I'd be happy to move them back to the .cc files if we can, but I according to what I read Clang is correct and we have to keep 'em in the headers.